partials/go.template

```diff
@@ -0,0 +1,111 @@
+{# defineGoType(schema)
+
+Input: schema
+Output: A Go type definition corresponding to the 
+        schema's data-type. Useful for declaring 
+        anonymous types and settings named fields
+#}
+{%- macro defineGoType(schema, tag) %}
+    {%- if schema.type() === "object" -%}
+        {{ asGoStruct(schema, tag) }}
+    {%- elseif schema.type() === "array" -%}
+        []{{ getGoType(schema.items()) }}
+    {%- else -%}
+        {{ schema.type() | toGoPrimitive(schema.format()) }}
+    {%- endif %}
+{%- endmacro %}
+
+{# getGoType(schema)
+
+    Wraps defineGoType() to return the schema's corresponding Go type identifier
+    if the schema was provided one in the AsyncAPI spec.
+
+Input: schema
+Output: A Go type derived from the schema's UID. 
+        If the schema was given an ID, it is converted to Go conventions.
+        If the schema was defined anonymously, defineGoType() is called 
+        to declare the type anonymously
+#}
+{%- macro getGoType(schema, tag) %}
+    {#- When not provided, AsyncAPI parser ID's anonymous schemas with '<anonymous-schema-#>' where # is a unique integer -#}
+    {%- if not schema.uid() | includes("anonymous-schema-") -%}
+        {{ schema.uid() | toGoPublicID }}
+    {%- else -%}
+        {{ defineGoType(schema, tag) }}
+    {%- endif %}
+{%- endmacro %}
+
+{# asGoStruct(schema)
+
+Input: schema
+Output: A Go struct definition
+#}
+{%- macro asGoStruct(schema, tag) -%}
+struct {
+{%- for propertyName, property in schema.properties() -%}
+    {%- set usePointer = "" if schema.required() | includes(propertyName) else "*" %}
+    // {{ propertyName | toGoPublicID }} is a property defined in the AsyncAPI specification
+    {{ propertyName | toGoPublicID }} {{usePointer}}{{ getGoType(property, tag) }}
+    {%- if tag %} `{{tag}}:"{{propertyName}}"`{%- endif %}
+{% endfor -%}
+}
+{%- endmacro %}
+
+{# messageName(schema)
+
+Input: schema
+Output: A Go type identifier of the message. 
+        Derived from the message's name field, if provided. Otherwise,
+        the message's UID is used.
+#}
+{%- macro messageName(msg) -%}
+    {%- set msgName = (msg.name() | toGoPublicID) if msg.name() else (msg.uid() | toGoPublicID) -%}
+    {{ msgName }}
+{%- endmacro -%}
+
+{# getOpBinding(operation)
+
+Input: operation
+Output: instantiation of a binding.{{protocol}} operation bindings struct
+
+#}
+{% macro getOpBinding(protocol, binding) %}
+    {% if protocol == "mqtt5" -%}
```

Comment:
is this optional and will without these bindings application work for any version of mqtt protocol?

also, you make a use of mqtt5 bindings here but they are not yet descrived here. would be great to get a PR here https://github.com/asyncapi/bindings/blob/master/mqtt5/README.md#operation

Reply:
This is optional. If no protocol binding is provided for an operation, this macro generates nothing. Will add this to the documentation.

Will look into working on a PR for the mqtt5 bindings. For now I was copying the mqtt ones. Not sure there's any difference between Mqtt & Mqtt5 in regards to the retain & QoS so those two should be straightforward enough.